src/charm.py

```diff
@@ -1,104 +1,397 @@
+            )
+            self._alertmanager_configurer_container.restart(
+                self._alertmanager_configurer_container_name
+            )
+            logger.info(f"Restarted container {self._alertmanager_configurer_service_name}")
+
+    def _start_dummy_http_server(self) -> None:
+        """Starts dummy HTTP server service.
+
+        Returns:
+            None
+        """
+        plan = self._dummy_http_server_container.get_plan()
+        layer = self._dummy_http_server_layer
+        if plan.services != layer.services:
+            self.unit.status = MaintenanceStatus(
+                f"Configuring pebble layer for {self._dummy_http_server_service_name}"
+            )
+            self._dummy_http_server_container.add_layer(
+                self._dummy_http_server_container_name, layer, combine=True
+            )
+            self._dummy_http_server_container.restart(self._dummy_http_server_service_name)
+            logger.info(f"Restarted container {self._dummy_http_server_service_name}")
+
+    def _on_alertmanager_config_changed(
+        self, event: Union[ConfigChangedEvent, AlertmanagerConfigFileChangedEvent]
+    ) -> None:
+        """Pushes configuration to Alertmanager through the relation data bag.
+
+        Args:
+            event: Juju ConfigChangedEvent or AlertmanagerConfigFileChangedEvent event
+
+        Returns:
+            None
+        """
+        if not self.model.unit.is_leader():
+            return
+        if not self.model.get_relation("alertmanager"):
+            self.unit.status = BlockedStatus("Waiting for alertmanager relation to be created")
+            event.defer()
+            return
+        if not self._alertmanager_config_file_exists:
+            self.unit.status = WaitingStatus(
+                "Waiting for Alertmanager config file to be available"
+            )
+            event.defer()
+            return
+        config = remote_config_write.load_config_file(self.ALERTMANAGER_CONFIG_FILE)
+        templates = self._get_templates(config)
+        self._update_alertmanager_relation_data_bag_with_new_alertmanager_config(config, templates)
+
+    def _get_templates(self, config: dict) -> list:
+        """Prepares templates data to be put in a relation data bag.
+        If the main config file contains templates section, content of the files specified in this
+        section will be concatenated. At the same time, templates section will be removed from
+        the main config, as alertmanager-k8s-operator charm doesn't tolerate it.
+        It is also possible to configure templates through the charm's templates_file config
+        option.
+        In case templates are specified in both the main config file and the charm's config, config
+        file will take precedence.
+
+        Args:
+            config: Alertmanager config
+
+        Returns:
+            list: List of templates
+        """
+        templates = []
+        if config.get("templates", []):
```

Comment:
The config name is `templates_file`

Reply:
Yes and it is handled lower. This line checks for the `templates` section in the config from the config file.

Reply:
Maybe outsourcing this check to a separate method will make it less confusing
```suggestion
        if _templates_section_in_config_file:
```